src/api/spec/controllers/webui/cloud/ec2/upload_job/logs_controller_spec.rb

```diff
@@ -0,0 +1,70 @@
+require 'rails_helper'
+require 'webmock/rspec'
+
+RSpec.describe Webui::Cloud::UploadJob::LogsController, type: :controller, vcr: true do
+  let!(:user) { create(:confirmed_user, login: 'adrian') }
+  let(:project) { create(:project, name: 'Apache') }
+  let!(:package) { create(:package, name: 'apache2', project: project) }
+
+  describe '#show' do
+    context 'with cloud_upload feature enabled' do
+      context 'without an EC2 configuration' do
+        before do
+          login(user)
+          Feature.run_with_activated(:cloud_upload) do
+            get :show, params: { upload_id: 42 }
+          end
+        end
+
+        it { expect(response).to redirect_to(cloud_ec2_configuration_path) }
+      end
+
+      context 'with an EC2 configuration' do
+        let!(:user_with_ec2_configuration) { create(:confirmed_user, login: 'tom') }
```

Comment:
I would create this like that:

```ruby
let(:ec2_configuration) { create(:ec2_configuration, user: user) }
let!(:user_with_ec2_configuration) { create(:confirmed_user, login: 'tom', ec2_configuration: ec2_configuration) }
```

conceptually creating a user called `user_with_ec2_configuration` without the configuration and creating it in the next line is a little bit confusing.

Reply:
That's way I asked for a factory before;-)